ote/src/clj/ote/services/transport.clj

```diff
@@ -247,17 +248,41 @@
                        (filter (comp (complement from-ui) ::t-service/id) from-db))]
     to-delete))
 
+(defn- delete-external-companies
+  "User might remove url from service, so we delete all service-companies from db"
+  [db transport-service]
+    (specql/delete! db ::t-service/service-company {::t-service/transport-service-id (::t-service/id transport-service)}))
+
+(defn- save-external-companies
+  "Service can contain an url that contains company nmes and business-id. Sevice can also contain an imported csv file
+  with company names and business-ids."
+  [db transport-service]
+  (let [current-data (first (fetch db ::t-service/service-company (specql/columns ::t-service/service-company)
+                            {::t-service/transport-service-id (::t-service/id transport-service)}))
+        companies (into [] (:companies (external/check-csv db {:url (get transport-service ::t-service/companies-csv-url)
```

Comment:
Destructure `::t-service/companies-csv-url` and `::t-service/id` to make this clearer.

Reply:
changed